### PR TITLE
[#543] Add EDI unit tests

### LINKED ISSUE #543: Add unit tests for EDI
We need to refactor and mock EDI loading procedure to allow it be unit-tested

### PR DESCRIPTION
**Pull request by kirillmakhonin**
This updates #543 
In this PR unit test skeleton for EDI and unit tests for EDI have been added

### CODE REVIEW

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test app

File: legion/tests/test_edi.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                models_info = edi.edi_client.inspect()
+                self.assertIsInstance(models_info, list)
+                self.assertEqual(len(models_info), 0)
+                self.assertTrue(len(get_models_patched.call_args_list) == 1, 'should be exactly one call')
+
+    def test_edi_inspect_model_id_and_version_empty(self):
+        TEST_MODEL_ID = 'test-id'
+        TEST_MODEL_VERSION = 'test-version'
+
+        with EDITestServer() as edi:
```
Comment (kirillmakhonin) [inline legion/tests/test_edi.py]: Try to move in to testSetup

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by ablatov** [comment]
bot, test app

**Comment by ablatov** [comment]
bot, test app

**Comment by ablatov** [comment]
bot, test app

**Comment by ablatov** [comment]
bot, test app

File: legion/tests/legion_test_utils.py [hunk 2]
```diff
@@ -83,6 +96,118 @@ def patch_environ(values, flush_existence=False):
     return patch('os.environ', new_values)
 
 
+@contextlib.contextmanager
+def persist_swagger_function_response_to_file(function, response_codes):
```
Comment (dsuslov) [inline legion/tests/legion_test_utils.py]: @kirillmakhonin do we really need this?

File: legion/tests/test_edi.py [hunk 3]
```diff
@@ -0,0 +1,568 @@
...
+class TestEDI(unittest2.TestCase):
+    def test_edi_inspect_all_detail(self):
+        with EDITestServer() as edi:
+            with m_func('kubernetes.client.CoreV1Api.list_namespaced_service', 'two_models_1_id_and_2_versions'), \
+                 m_func('kubernetes.client.ExtensionsV1beta1Api.list_namespaced_deployment',
+                        'two_models_1_id_and_2_versions'), \
+                 mock.patch('legion.k8s.utils.build_client', return_value=None):
+                models_info = edi.edi_client.inspect()
+                # Test count of returned models
+                self.assertIsInstance(models_info, list)
```
Comment (dsuslov) [inline legion/tests/test_edi.py]: @ablatov please, store expected responses as text (json or other format) in test resources folder and load them in tests to compare dicts of actual and expected response.

File: legion/tests/legion_test_utils.py (hunk 2 above)
Comment (kirillmakhonin) [reply legion/tests/legion_test_utils.py]: sure, I'll simplify this function and remove possibility to pass response_codes as a list. Also argument name can be changed, but I'm not sure which will be correct.

**Comment by kirillmakhonin** [comment]
@dsuslov please re-review
Tests will be started when develop branch became working

**Comment by ablatov** [comment]
bot, test app

**Comment by ablatov** [comment]
bot, test app

**Pull request merged**